### PR TITLE
NativeAOT-LLVM: Pass this to calls - initial support for GT_IND and GT_NE

### PR DESCRIPTION
**Pull request by yowl**
This PR adds support for passing `this` to instance methods from the clrjit->LLVM module.  Also starts support for GT_IND and GT_NE.  I moved a few methods around to prevent forward references, so sorry about that churn.

### CODE REVIEW

File: src/coreclr/jit/llvm.cpp [hunk 1]
```diff
@@ -64,6 +65,8 @@ extern "C" DLLEXPORT void registerLlvmCallbacks(void*       thisPtr,
         _module = new Module(llvm::StringRef("netscripten-clrjit"), _llvmContext);
         _module->setTargetTriple(triple);
         _module->setDataLayout(dataLayout);
+        // TODO: better to get this from compiler, but where is it?
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]: I don't think we have a dynamic value for it. Forms I've personally seen in the past: `TARGET_POINTER_SIZE` define, `genTypeSize(TYP_I_IMPL)`. Seems like the define is most popular.

File: src/coreclr/jit/llvm.cpp [hunk 2]
```diff
@@ -235,8 +283,65 @@ Value* genTreeAsLlvmType(GenTree* tree, Type* type)
     failFunctionCompilation();
 }
 
-llvm::Value* buildUserFuncCall(GenTreeCall* gtCall, llvm::IRBuilder<>& builder)
+int getTotalParameterOffset(CORINFO_SIG_INFO &sigInfo)
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]:
```suggestion
int getTotalParameterOffset(CORINFO_SIG_INFO& sigInfo)
```

File: src/coreclr/jit/llvm.cpp [hunk 3]
```diff
@@ -235,8 +283,65 @@ Value* genTreeAsLlvmType(GenTree* tree, Type* type)
...
+{
+    if (offset == 0)
+        return shadowStack;
+
+    return _builder->CreateGEP(shadowStack, _builder->getInt32(_pointerSize));
+}
+
+bool isThisArg(GenTreeCall* gtCall, GenTree* operand)
 {
+    // TODO: is the this arg always a late arg?
```
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]: I think you can make this logic not dependent on the lateness of `this`:
```c++
if (gtCall->gtCallThisArg == nullptr)
{
    return false;
}
   
return _compiler->gtGetThisArg(gtCall) == operand;
```
Also nit: `gtCall` -> `call` (here and elsewhere). The purpose of `gt`/`fg`/`gen` and others is the namespacing of sorts for when everything is a `Compiler` member. I don't think there is need to use the prefixes for locals.

Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: Much better, thanks.

File: src/coreclr/jit/llvm.cpp (hunk 2 above)
Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: Done.

**Comment by yowl** [comment]
Having problems with the CI...

**Comment by SingleAccretion** [comment]
The CI <-> GitHub flow has broken recently, but the pipeline is still running. See https://dev.azure.com/dnceng/public/_build?definitionId=841.

**Review by jkotas — APPROVED** [review]
LGTM

**Pull request merged**